examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java

```diff
@@ -258,6 +263,108 @@ void createOtherGroup () {
 		resizableColumns.addSelectionListener (widgetSelectedAdapter(event -> setColumnsResizable ()));
 		headerImagesButton.addSelectionListener (widgetSelectedAdapter(event -> recreateExampleWidgets ()));
 		subImagesButton.addSelectionListener (widgetSelectedAdapter(event -> recreateExampleWidgets ()));
+		editableButton.addSelectionListener(widgetSelectedAdapter(event -> makeTreeContentEditable()));
+	}
+
+	/**
+	 * Make Tree Editable
+	 */
+	void makeTreeContentEditable() {
+		// Tree 1
+		final TreeEditor editor = new TreeEditor(tree1);
+		editor.horizontalAlignment = SWT.LEFT;
+		editor.grabHorizontal = true;
+		// Add double click listener to the tree
+		tree1.addListener(SWT.MouseDoubleClick, new Listener() {
```

Comment:
`Listener` is a functional interface, so you can use a lambda instead of providing an anonymous implementation of that interface.